dashboard/src/views/SystemSpec.vue

```diff
@@ -0,0 +1,218 @@
+        <br />
+        <b>Temperature Model:</b>
+        <input disabled v-model="modelSpec.temperature_model" />
+        <br />
+      </div>
+
+      <div>
+        <system-view :system="system" :model="model" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from "vue-property-decorator";
+
+import { System } from "@/types/System";
+import { modelSpecs } from "@/types/ModelSpecification";
+import {
+  SandiaInverterParameters,
+  PVWattsInverterParameters
+} from "@/types/InverterParameters";
+
+import SystemView from "@/components/model/System.vue";
+import FileUpload from "@/components/FileUpload.vue";
+
+Vue.component("system-view", SystemView);
+Vue.component("file-upload", FileUpload);
+
+@Component
+export default class SystemSpec extends Vue {
+  @Prop({ default: null }) systemId!: number | null;
+  system!: System;
+  model!: string;
+  loading!: boolean;
+  errorState!: boolean;
+
+  created() {
+    if (this.systemId != undefined) {
+      this.loadSystem();
+    } else {
+      this.system = new System({});
+      this.loading = false;
+    }
+  }
+  data() {
+    return {
+      system: this.system ? this.system : new System({}),
+      modelPresetOptions: ["pvsyst", "pvwatts"],
+      model: "pvsyst",
+      displaySummary: false,
+      displayAdvanced: false,
+      loading: true,
+      errorState: false
+    };
+  }
+  components = ["system-view", "file-upload"];
+  uploadSuccess(fileMetadata: string) {
+    const metadata = JSON.parse(fileMetadata);
+    const system = new System(metadata);
+    this.system = system;
+    this.inferModel();
+  }
+
+  downloadSystem() {
+    const contents = new Blob([JSON.stringify(this.system, null, 2)], {
+      type: "application/json;charset=utf-8;"
+    });
+    const filename = `${this.system.name}.json`;
+    if (navigator.msSaveBlob) {
+      navigator.msSaveBlob(contents, filename);
+    } else {
+      const link = document.createElement("a");
+      link.href = URL.createObjectURL(contents);
+      link.download = filename;
+      link.target = "_blank";
+      link.style.visibility = "hidden";
+      link.dispatchEvent(new MouseEvent("click"));
+      link.remove();
+    }
+  }
+
+  get modelSpec() {
+    return modelSpecs[this.model];
+  }
+
+  inferModel() {
+    if (this.system.inverters.length > 0) {
+      const firstParams = this.system.inverters[0].inverter_parameters;
+
+      if (SandiaInverterParameters.isInstance(firstParams)) {
+        this.model = "pvsyst";
+      } else if (PVWattsInverterParameters.isInstance(firstParams)) {
+        this.model = "pvwatts";
+      }
+    }
+  }
+
+  async loadSystem() {
+    const token = await this.$auth.getTokenSilently();
+    const response = await fetch(`/api/systems/${this.systemId}`, {
+      headers: new Headers({
+        Authorization: `Bearer ${token}`
+      })
+    });
+    if (response.ok) {
+      const system: System = await response.json();
```

Comment:
this is really a StoredSystem. So for now the system should be the .defnition attribute of the returned json